gitlab/resource_gitlab_service_pipelines_email.go

```diff
@@ -0,0 +1,103 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabServicePipelinesEmail() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabServicePipelinesEmailCreate,
+		Read:   resourceGitlabServicePipelinesEmailRead,
+		Update: resourceGitlabServicePipelinesEmailCreate,
+		Delete: resourceGitlabServicePipelinesEmailDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabServicePipelinesEmailImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"recipients": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"notify_only_broken_pipelines": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"branches_to_be_notified": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"all", "default", "protected", "default_and_protected"}, true),
+				Default:      "default",
+			},
+		},
+	}
+}
+
+func resourceGitlabServicePipelinesEmailSetToState(d *schema.ResourceData, service *gitlab.PipelinesEmailService) {
+	d.SetId(fmt.Sprintf("%d", service.ID))
+	d.Set("recipients", strings.Split(service.Properties.Recipients, ","))
+	d.Set("notify_only_broken_pipelines", service.Properties.NotifyOnlyBrokenPipelines)
+	d.Set("branches_to_be_notified", service.Properties.BranchesToBeNotified)
+}
+
+func resourceGitlabServicePipelinesEmailCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	options := &gitlab.SetPipelinesEmailServiceOptions{
+		Recipients:                gitlab.String(strings.Join(*stringSetToStringSlice(d.Get("recipients").(*schema.Set)), ",")),
+		NotifyOnlyBrokenPipelines: gitlab.Bool(d.Get("notify_only_broken_pipelines").(bool)),
+		BranchesToBeNotified:      gitlab.String(d.Get("branches_to_be_notified").(string)),
+	}
+
+	log.Printf("[DEBUG] create gitlab pipelines emails service for project %s", project)
+
+	_, err := client.Services.SetPipelinesEmailService(project, options)
+	if err != nil {
+		return err
+	}
+
+	return resourceGitlabServicePipelinesEmailRead(d, meta)
+}
+
+func resourceGitlabServicePipelinesEmailRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
```

Comment:
Can you change this to

```suggestion
	project := d.Id()
```

so you don't need the importer?

Idea from here - https://www.terraform.io/docs/extend/resources/import.html#importer-state-function
> Sometimes it is possible to adjust the resource Read function to replace d.Get() use with d.Id() if they exactly match